sdk/include/opentelemetry/sdk/metrics/exemplar/filtered_exemplar_reservoir.h

```diff
@@ -0,0 +1,65 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+#ifndef ENABLE_METRICS_PREVIEW
+#  include <memory>
+#  include <vector>
+#  include "opentelemetry/context/context.h"
+#  include "opentelemetry/nostd/shared_ptr.h"
+#  include "opentelemetry/sdk/common/attribute_utils.h"
+#  include "opentelemetry/sdk/metrics/exemplar/filter.h"
+#  include "opentelemetry/sdk/metrics/exemplar/reservoir.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace metrics
+{
+class FilteredExemplarReservoir final : public ExemplarReservoir
```

Comment:
as per the [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#exemplar-defaults), there are only two reservoirs -SimpleFixedSizeExemplarReservoir, AlignedHistogramBucketExemplarReservoir. What is the purpose of this reservoir.

Reply:
This is a reservoir that has a pre-filter on measurements. Java has it. Shall I remove it?
I think a filter per trace would be interesting for the users.